lib/govuk_test.rb

```diff
@@ -5,16 +5,24 @@
 require "puma"
 require "selenium-webdriver"
 
+unless ENV['GOVUK_TEST_USE_SYSTEM_CHROMEDRIVER']
+  require 'webdrivers'
+
+  # This stop webdrivers doing a lazy check for new versions
+  # of chromedriver, which interferes with WebMock.disable_net_connect
+  Selenium::WebDriver::Chrome.driver_path
+end
+
 module GovukTest
   def self.configure(options = {})
     chrome_options = %w(headless disable-gpu)
     chrome_options << "--window-size=#{options[:window_size]}" if options[:window_size]
 
-    if ENV['GOVUK_TEST_USE_SYSTEM_CHROMEDRIVER']
-      # Use the installed chromedriver, rather than chromedriver-helper
-      Selenium::WebDriver::Chrome.driver_path = File.which("chromedriver")
-    else
-      require 'chromedriver-helper'
+    if defined?(WebMock)
```

Comment:
Do we still need these block?

Reply:
I believe that not every `govuk_test` using app includes WebMock.
https://github.com/alphagov/publishing-e2e-tests/blob/master/Gemfile.lock
So until we make `govuk_test` the WebMock configurator in a future PR this is probably still needed

Reply:
I'm a little confused now - why do we need to add this code?

Reply:
using webdrivers causes an issue with WebMock, which this block resolves
https://github.com/titusfortner/webdrivers/issues/4

Reply:
Ah, so I think something's got lost in communication here (or maybe there's something I haven't considered). The lines 7-13 should make the WebMock thing redundant, which is preferable as it means we can avoid a breaking change. I wrote a comment to try and explain this:

> I've now had some time to play myself, and can confirm the following works. We do need to make sure it comes before any call to `disable_net_connect`, and because these lines don't actually configure anything, it seems reasonable to achieve this by doing them outside of `.configure`.
> 
> ```
> require "selenium-webdriver"
> 
> unless ENV['GOVUK_TEST_USE_SYSTEM_CHROMEDRIVER']
>   require 'webdrivers'
> 
>   # This stop webdrivers doing a lazy check for new versions
>   # of chromedriver, which interferes with WebMock.disable_net_connect
>   Selenium::WebDriver::Chrome.driver_path
> end
> 
> module GovukTest
>   def self.configure(options = {})
>     chrome_options = %w(headless disable-gpu)
>     chrome_options << "--window-size=#{options[:window_size]}" if options[:window_size]
> 
>     #if defined?(WebMock)
>       #WebMock.disable_net_connect!(
>         #allow_localhost: true,
>         #allow: 'chromedriver.storage.googleapis.com'
>       #)
>     #end
> ```
> This should mean we can avoid a breaking change, although it still sounds like a good idea to abstract the setup of WebMock into this gem.

Reply:
@edwardkerry pointed out this doesn't work in Content Tagger, as webdrivers takes advantage of the newer version of the selenium gem to actually do the lazy behaviour. The following fixes it.

```
  # This stops webdrivers doing a lazy check for new versions
  # of chromedriver, which interferes with WebMock.disable_net_connect
  if ::Selenium::WebDriver::Service.respond_to? :driver_path=
    Selenium::WebDriver::Chrome::Service.driver_path = Webdrivers::Chromedriver.update                                                                                                                                                                                        
  else
    Selenium::WebDriver::Chrome.driver_path = Webdrivers::Chromedriver.update
  end
```

This overrides the behaviour of the `webdrivers/selenium` helper, which has to cope with a deprecated API on the selenium gem, hence the `if` statement.

Reply:
On the positive side, it means we're just using webdrivers to manage chromedriver and doing (overriding) the actual setup of the Selenium driver ourselves #nomoremagic.